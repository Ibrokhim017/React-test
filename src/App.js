import React, { useEffect, useState } from "react";

const App = () => {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState("");

  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  }, []);

  const onClick = () => setToggle((prev) => !prev);

  return (
    <div>
      <h2 data-testId="value-elem">{value}</h2>

      {data && <h2 style={{ color: "red" }}>Data</h2>}

      {toggle && <div data-testId="toggle-elem">toggle</div>}

      <h1>Hello world</h1>

      <button data-testId="toggle-btn" onClick={onClick}>
        Click me
      </button>

      <input
        type="text"
        placeholder="value"
        value={value}
        onChange={(e) => setValue(e.target.value)}
      />
    </div>
  );
};

export default App;
