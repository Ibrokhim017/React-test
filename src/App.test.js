import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("REACT TEST", () => {
  test("getBy text, role, placeholder", () => {
    render(<App />);

    const helloWorldElem = screen.getByText(/click me/i);
    const btn = screen.getByRole(/button/i);
    const input = screen.getByPlaceholderText(/value/i);

    expect(helloWorldElem).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toMatchSnapshot();

    screen.debug();
  });

  test("FindBy Async test", async () => {
    render(<App />);

    screen.debug();
    const dataText = await screen.findByText(/data/i);
    expect(dataText).toBeInTheDocument();

    expect(dataText).toHaveStyle({ color: "red" });

    screen.debug();
  });

  test("queryBy Toggle test", () => {
    render(<App />);

    screen.debug();
    const btn = screen.getByTestId("toggle-btn");
    expect(screen.queryByTestId("toggle-elem")).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle-elem")).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle-elem")).toBeNull();
    screen.debug();
  });

  test("INPUT EVENT", () => {
    render(<App />);

    const input = screen.getByPlaceholderText(/value/i);
    expect(screen.queryByTestId("value-elem")).toContainHTML("");

    // fireEvent.input(input, {
    //   target: {
    //     value: "123123",
    //   },
    // });

    // userEvent huddi react eventlar kabiishalaydi
    userEvent.type(input, "123123");

    expect(screen.queryByTestId("value-elem")).toContainHTML("123123");
  });
});
