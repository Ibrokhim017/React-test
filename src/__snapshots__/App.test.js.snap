// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REACT TEST getBy text, role, placeholder 1`] = `
<input
  placeholder="value"
  type="text"
  value=""
/>
`;
